lib/compiler/src/beam_ssa_pre_codegen.erl

```diff
@@ -1841,24 +1841,55 @@ find_update_succ([S|Ss], #dk{d=Defs0,k=Killed0}=DK0, D0) ->
 find_update_succ([], _, D) -> D.
 
 find_yregs_is([#b_set{dst=Dst}=I|Is], #dk{d=Defs0,k=Killed0}=Ys, Yregs0) ->
-    Used = beam_ssa:used(I),
-    Yregs1 = ordsets:intersection(Used, Killed0),
-    Yregs = ordsets:union(Yregs0, Yregs1),
+    Yregs1 = intersect_used(I, Killed0),
+    Yregs = cerl_sets:union(Yregs0, Yregs1),
     case beam_ssa:clobbers_xregs(I) of
         false ->
             Defs = ordsets:add_element(Dst, Defs0),
             find_yregs_is(Is, Ys#dk{d=Defs}, Yregs);
         true ->
-            Killed = ordsets:union(Defs0, Killed0),
+            Killed = cerl_sets:union(cerl_sets:from_list(Defs0), Killed0),
             Defs = [Dst],
             find_yregs_is(Is, Ys#dk{d=Defs,k=Killed}, Yregs)
     end;
 find_yregs_is([], Ys, Yregs) -> {Yregs,Ys}.
 
 find_yregs_terminator(Terminator, #dk{k=Killed}, Yregs0) ->
-    Used = beam_ssa:used(Terminator),
-    Yregs = ordsets:intersection(Used, Killed),
-    ordsets:union(Yregs0, Yregs).
+    Yregs = intersect_used(Terminator, Killed),
+    cerl_sets:union(Yregs0, Yregs).
+
+intersect_used(#b_blk{is=Is,last=Last}, Set) ->
```

Comment:
The main idea of this patch is replacing `beam_ssa:used+intersection` by this new `intersect_used` with computes the "intersect used" values by checking against a `cerl_sets` at the same time we traverse.

Question: do we want to move `intersect_used` to beam_ssa? The only benefit is that we won't have to duplicate `beam_ssa:used_args/1` below?

Reply:
`beam_ssa` is meant for functions that are used from more than one module, or are expected to be used from other modules in the future. If you don't think that it can be used from other modules, I would suggest that you simplify it and keep in place.